ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@lukewagner, with tuples, the nullary case would be covered by the nullary tuple, and `br0` should then behave like a shorthand for `(br (tuple))`, me thinks. That only works if you generalise from syntactic to semantic arity matching, which is useful for other reasons. Also, I'm not sure you actually want to treat tuples as single entries on the expression stack, but rather as multiple entries.

But either way, right now my main concern is not to lock us in.